invalid assert in lib/command.c (found with gcc 7)

(stable/3.0 branch - not affecting master)

It seems `lib/command.c` has an invalid assert. From gcc 7 warning:
```command.c: In function ‘cmd_merge_nodes’:
command.c:517:12: error: ?: using integer constants in boolean context, the expression will always evaluate to ‘true’ [-Werror=int-in-bool-context]
			: 1);
./zassert.h:38:11: note: in definition of macro ‘zassert’
  ((void)((EX) ? 0 : (_zlog_assert_failed(#EX, __FILE__, __LINE__,       \
		   ^~
command.c:514:4: note: in expansion of macro ‘assert’
	assert(vector_count(cnew->from)
	^~~~~~
```

Which matches this section in `lib/command.c`:
```if (i == vector_active(old->to) && direction > 0) {
	assert(vector_count(cnew->from)
						   == cmd_nodes_link(cnew, cnew)
				   ? 2
				   : 1);
	graph_remove_edge(new, cnew);

	cmd_reparent_tree(newgraph, oldgraph, cnew);

	graph_add_edge(old, cnew);
}
```

The contents of the assert is always 1 or 2, so always true -> never triggering the assert.

Assigning to @qlyoung on his request (Thanks!)

This affects blocks issue #1469 for stable/3.0

--- a/lib/command.c
+++ b/lib/command.c
@@ -511,10 +511,6 @@ static void cmd_merge_nodes(struct graph *oldgraph, struct graph *newgraph,
 		}
 		/* nothing found => add new to old */
 		if (i == vector_active(old->to) && direction > 0) {
-			assert(vector_count(cnew->from)
-					       == cmd_nodes_link(cnew, cnew)
-				       ? 2
-				       : 1);
 			graph_remove_edge(new, cnew);
 
 			cmd_reparent_tree(newgraph, oldgraph, cnew);

